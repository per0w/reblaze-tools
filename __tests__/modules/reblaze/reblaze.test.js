const yargs = require('yargs');
/* eslint-disable no-unused-expressions */
yargs(['reblaze:hostDisable', 'host:server1,server2,server3,server4']).default('site', 'site.com')
    .argv;

const reblaze = require('../../../app/modules/reblaze/reblaze');

describe('Reblaze', () => {
    const fServers = reblaze.findServers();
    it('findServers data correctly', () => {
        expect(fServers).toMatchSnapshot();
    });
    it('createConfig when server is disable', async () => {
        const spy = jest.spyOn(reblaze, 'createConfig');
        const createConfig = await reblaze.createConfig(true);
        expect(spy).toHaveBeenCalled();
        expect(createConfig).toMatchSnapshot();
    });
    it('createConfig when server is enable', async () => {
        const spy = jest.spyOn(reblaze, 'createConfig');
        const createConfig = await reblaze.createConfig(false);
        expect(spy).toHaveBeenCalled();
        expect(createConfig).toMatchSnapshot();
    });
});
