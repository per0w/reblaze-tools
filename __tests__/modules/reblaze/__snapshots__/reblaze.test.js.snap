// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reblaze createConfig when server is disable 1`] = `
Array [
  Object {
    "backup": false,
    "down": true,
    "fail_timeout": "10s",
    "host": "35.225.62.170",
    "http_port": "80",
    "https_port": "443",
    "max_fails": "1",
    "monitor_state": "0",
    "weight": "1",
  },
  Object {
    "backup": false,
    "down": true,
    "fail_timeout": "10s",
    "host": "35.224.181.95",
    "http_port": "80",
    "https_port": "443",
    "max_fails": "1",
    "monitor_state": "0",
    "weight": "1",
  },
]
`;

exports[`Reblaze createConfig when server is enable 1`] = `
Array [
  Object {
    "backup": false,
    "down": false,
    "fail_timeout": "10s",
    "host": "35.225.62.170",
    "http_port": "80",
    "https_port": "443",
    "max_fails": "1",
    "monitor_state": "0",
    "weight": "1",
  },
  Object {
    "backup": false,
    "down": false,
    "fail_timeout": "10s",
    "host": "35.224.181.95",
    "http_port": "80",
    "https_port": "443",
    "max_fails": "1",
    "monitor_state": "0",
    "weight": "1",
  },
]
`;

exports[`Reblaze findServers data correctly 1`] = `
Object {
  "35.193.224.71": "node4",
  "35.224.128.225": "node3",
  "35.224.181.95": "node2",
  "35.225.24.224": "admin",
  "35.225.62.170": "node1",
  "35.226.19.133": "node5",
}
`;
